src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerMetrics.java

```diff
@@ -242,4 +174,145 @@ private void addLatency(JsonObject json, Histogram histogram, String prefix) {
       .put("99.5", histogram.getValueAtPercentile(99.5))
       .put("100", histogram.getValueAtPercentile(100)));
   }
+
+  private static class RollingWindow {
+    private final Summary history;
+    private final Summary[] buckets;
+    private final long bucketSizeInNs;
+
+    RollingWindow(long windowSizeInMs, int numberOfBuckets) {
+      if (windowSizeInMs % numberOfBuckets != 0) {
+        throw new IllegalArgumentException("Window size should be divisible by number of buckets.");
+      }
+      this.buckets = new Summary[numberOfBuckets];
+      for (int i = 0; i < buckets.length; i++) {
+        this.buckets[i] = new Summary();
+      }
+      this.bucketSizeInNs = 1_000_000 * windowSizeInMs / numberOfBuckets;
+      this.history = new Summary();
+    }
+
+    public void add(Operation operation) {
+      getBucket(operation.end).add(operation);
```

Comment:
In the original implementation, one looked at the begin timestamps:
```
window.removeIf(operation -> operation.begin < beginningOfTheWindow);
```
This causes operations which take more than 10 seconds to not being taken into account in the rolling statistics.  So that is why I decided to look at `.end`.